### DISCUSSION TITLE
Wav file input for n speakers;    n Wav files output each with 1 diarised speaker

### DISCUSSION THREAD

**Discussion by rleaver152** [pyannote/pyannote-audio#1214] [Q&A]
Apologies if a FAQ but I've been going through everything and cannot see how to accomplish a python script which:

1) Wav file input for n speakers
2) uses pyannote-audio to (in a pipeline) output  n individual Wav files each with 1 diarised speaker,  each file of the same length.

So if you merged the n Wav output files you would end up with something very like the original Wav file input.   The objective being to have individual speakers' wav files.

**Comment by hbredin** [comment]
See related issue: https://github.com/pyannote/pyannote-audio/issues/1014

**Reply by rleaver152** [reply]
Many thanks.   If I understand correctly,   I can get <speech_start_label1> and <speech_end_label1> sets of timestamps for speaker label1 etc etc.     

Do you know if anyone has done a script to take these and implement  ffmpeg snips of the original audio to output speaker label1... etc output only?   I don't mind overlap cases,  just keeping it simple as proof of concept.

I know how to seek and cut and merge wav sequences with ffmpeg and also introduce silences using ffmpeg.    So my dummy script would address in a source file 00:00:00 to 01:00:00 :

label1    00:00:05     00:00:10
label2    00:00:15     00:00:20

resulting in a a single output WAV using ffmpeg which has 
silence from 00:00:00 to 00:00:05,  00:00:10 to 00:00:15  and 00:00:20 to 01:00:00
label1 speaker 00:00:05 to 00:00:10 
label 2 speaker 00:00:15 to 00:00:20 

Whilst I can write this in a bash script,  I don't really want to reinvent the wheel since it seems such an obvious requirement for the community so keen to know if anyone has done it before.

Obviously as described above pretty much ends up with the input (except for enforced non-speaking silence) to give clarity in writing it down here,  but the idea is I can then enable it only for each individual speaker separately so I end up with a WAV file of identical length for each label speaker.
